custom_components/omnik_inverter/config_flow.py

```diff
@@ -287,20 +293,64 @@ async def async_step_init(
         Returns:
             The created config entry.
         """
+        errors = {}
+
         if user_input is not None:
-            return self.async_create_entry(title="", data=user_input)
+            try:
+                await validate_input(user_input)
+            except OmnikInverterError:
+                LOGGER.exception("Failed to connect to the Omnik")
+                errors["base"] = "cannot_connect"
+            except Exception as error:  # pylint: disable=broad-except
+                errors["base"] = str(error)
+            else:
+                updated_config = {CONF_SOURCE_TYPE: self.source_type}
+                for key in (CONF_HOST, CONF_USERNAME, CONF_PASSWORD, CONF_SERIAL):
+                    if key in user_input:
+                        updated_config[key] = user_input[key]
+
+                new_title = user_input.get(CONF_NAME)
+                if new_title:
+                    self.hass.config_entries.async_update_entry(
+                        self.config_entry,
+                        data=updated_config,
+                        title=new_title,
+                    )
+                else:
+                    self.hass.config_entries.async_update_entry(
+                        self.config_entry,
+                        data=updated_config,
+                    )
+
+                options = {}
+                for key in (CONF_SCAN_INTERVAL, CONF_USE_CACHE):
+                    options[key] = user_input[key]
+                return self.async_create_entry(title="", data=options)
+
+        fields = {
+                vol.Optional(
+                    CONF_NAME,
```

Comment:
Why not using a default here? 
```suggestion
                    CONF_NAME, default=self.config_entry.data.get(CONF_NAME)
```
You may then no longer need the If statement to determine a "new_title".

Reply:
Is because if default is used, and user don't want to change the actual name, has to reenter the name. I was not able to find out how to retrieve the actual title of the integration object. It is not stored in the config data. You know maybe? The could do as you suggested. Was also my first intention to do like that.

Reply:
I got it how to do that, will change it

Reply:
done